File: Chapter-05-Linked-Lists/SList-Avg/SList-Avg.py
# Create a function average(head) to return average of the lists's values
# assume all entries are numeric

# imort the list class
import sys
from os.path import abspath, dirname

mainDir = dirname(dirname(abspath(__file__)))
sys.path.append(mainDir)

from singleList import List

# import the random module
from random import randrange

# create a new list and populate it with random numbers
newSList = List()
for i in range(0, 10):
    newSList.add(randrange(100))

# function to get average value
def average(head):
    sum = 0
    len = 0
    if head:
        while head:
            sum += head.val
            len += 1
            head = head.next
        return sum/len
    else:
        return 'The list is empty'

# print the list
print('The list is {}').format(newSList.show())

# print the average value
print('The average of values in the list is {}').format(average(newSList.getHeadNode()))

File: Chapter-05-Linked-Lists/SList-Avg/SList-Avg.js
// Create a function average(head) to return average of the  lists's values
// assume all entries are numeric

// import the list class
var path = require('path'),
    SList= require(path.join(__dirname, '../singleList'))

// create a new list and populate it with random numbers
var newSList = new SList()
for (let i = 0; i < 10; i++){
  newSList.add(Math.floor(Math.random() * 100))
}

// function to get average value
function average(head){
  let sum = 0,
      len = 0
  if (head){
    while (head){
      sum += head.val
      len += 1
      head = head.next
    }
    return (sum / len)
  } else {
    return `The list is empty`
  }
}

// print the list
console.log(`The list is ${newSList.show()}`)

// print the average value
console.log(`The average of values in the list is ${average(newSList.getHeadNode())}`)
